legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -253,8 +279,19 @@ private void sendResponse(final RestChannel channel, final String message, final
         channel.sendResponse(new BytesRestResponse(status, message));
     }
 
-    private void reportError(final RestChannel channel, final Exception e, final RestStatus status) {
-        sendResponse(channel, ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString(), status);
+    /**
+     * Report Error message to user.
+     * @param channel : Rest channel to sent response through.
+     * @param e : Exception caught when attempting query.
+     * @param status : Status for rest request made.
+     * @param v2SqlEngineError : Error message for new SQL engine. Can be removed when old SQL engine is deprecated.
+     */
+    private void reportError(final RestChannel channel, final Exception e, final RestStatus status, String v2SqlEngineError) {
+        String errorMsg = ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString();
+        errorMsg += v2SqlEngineError.isEmpty() ? "" :
+            "\nQuery failed on both V1 and V2 SQL parser engines. V2 SQL parser error following: \n"
+            + v2SqlEngineError;
```

Comment:
Just discovered that you're concatenating 2 json objects here. In my test I'm getting
```
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "unsupported method: simple_query_string",
    "type": "SqlParseException"
  },
  "status": 400
}
Query failed on both V1 and V2 SQL parser engines. V2 SQL parser error following:
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "Failed to parse query due to offending symbol [1] at: 'select `key`, str2 from calcs where simple_query_string([str2], one, analyzer = standard, 1' <--- HERE... More details: Expecting tokens in {'ALLOW_LEADING_WILDCARD', 'ANALYZER', 'ANALYZE_WILDCARD', 'AUTO_GENERATE_SYNONYMS_PHRASE_QUERY', 'BOOST', 'CUTOFF_FREQUENCY', 'DEFAULT_FIELD', 'DEFAULT_OPERATOR', 'ESCAPE', 'ENABLE_POSITION_INCREMENTS', 'FIELDS', 'FLAGS', 'FUZZINESS', 'FUZZY_MAX_EXPANSIONS', 'FUZZY_PREFIX_LENGTH', 'FUZZY_REWRITE', 'FUZZY_TRANSPOSITIONS', 'LENIENT', 'LOW_FREQ_OPERATOR', 'MAX_DETERMINIZED_STATES', 'MAX_EXPANSIONS', 'MINIMUM_SHOULD_MATCH', 'OPERATOR', 'PHRASE_SLOP', 'PREFIX_LENGTH', 'QUOTE_ANALYZER', 'QUOTE_FIELD_SUFFIX', 'REWRITE', 'SLOP', 'TIE_BREAKER', 'TIME_ZONE', 'TYPE', 'ZERO_TERMS_QUERY'}",
    "type": "SyntaxCheckException"
  },
  "status": 400
}
```
You should update `details` block of `ErrorMessage` instead.